### PR TITLE
add test with jasmine and fix some issue

### PR DESCRIPTION
**Pull request by shuizhongyueming**
. add jasmine to make this project testable.
. fix the result of stringType for null and undefined
. fix the result of Type.of(Math) and Type.of(JSON) to Be `[Function: Object]`

### CODE REVIEW

File: spec/spec.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+describe('type-of-is', function(){
```
Comment (stephenhandley) [inline spec/spec.js]: this doesn't improve test coverage it just changes the syntax. not interested in including this.

File: README.md [hunk 2]
```diff
@@ -157,8 +157,8 @@ console.log(Type.instance(ralph, Object));   // true
 console.log(Type.of(Infinity));    // [Function: Number]
 console.log(Type.of(-Infinity));   // [Function: Number]
 console.log(Type.of(NaN));         // [Function: Number]
-console.log(Type.of(Math));        // {}
-console.log(Type.of(JSON));        // {}
+console.log(Type.of(Math));        // [Function: Object]
```
Comment (stephenhandley) [inline README.md]: yep sounds good, thx

File: index.js [hunk 3]
```diff
@@ -37,9 +37,9 @@
       // For now work around this bug in PhantomJS
       // https://github.com/ariya/phantomjs/issues/11722
```
Comment (stephenhandley) [inline index.js]: this issue now appears to be fixed / closed in phantom. I think the better approach is to revert these temporary changes https://github.com/stephenhandley/type-of-is/commit/a22f0fca9ae1edf5c15baa7fe85ea3b3aba0c768

File: spec/spec.js (hunk 1 above)
Comment (shuizhongyueming) [inline spec/spec.js]: Yes,  with console.log you can get the result of every statements, but you don't know whether the result is what you want. 

And with jasmine, every result will be compare with the result you want automatically.
image

I think this is really important for the stable of this project.

**Pull request closed without merging**